src/main/java/org/opensearch/geospatial/ip2geo/action/UpdateDatasourceTransportAction.java

```diff
@@ -94,6 +95,11 @@ protected void doExecute(final Task task, final UpdateDatasourceRequest request,
                         if (datasource == null) {
                             throw new ResourceNotFoundException("no such datasource exist");
                         }
+                        if (DatasourceState.AVAILABLE.equals(datasource.getState()) == false) {
```

Comment:
Why not use if(!DatasourceState.....)?

Reply:
`== false` is more readable and preferred format borrowed from OpenSearch core.